crates/forge_main/src/ui.rs

```diff
@@ -848,6 +851,26 @@ impl<A: API + 'static, F: Fn() -> A> UI<A, F> {
         Ok(())
     }
 
+    async fn on_completion(&mut self, summary: SessionSummary) -> anyhow::Result<()> {
+        self.spinner.stop(None)?;
+
+        // Show usage info
+        self.writeln(Info::from(&self.state))?;
+        // Show summary
+        self.writeln(Info::from(summary))?;
+
+        let should_start_new_chat = ForgeSelect::confirm("Do you want to start a new chat?")
+            .with_default(true)
+            .prompt()?;
+
+        // if conversation is over
+        if should_start_new_chat.unwrap_or(false) {
+            self.on_new().await?;
+        }
+
+        Ok(())
```

Comment:
```
SESSION SUMMARY
Duration: 1m 30s
Lines: -76 +413

? Task completed. Start a new chat? (Y/n)
```
Can we make the final output look like above.

Reply:
yes, ofc!